server/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -703,6 +708,29 @@ public interface BlockLoaderContext {
          * The {@code _field_names} field mapper, mostly used to check if it is enabled.
          */
         FieldNamesFieldMapper.FieldNamesFieldType fieldNames();
+
+        @Nullable
+        default BlockLoaderValueFunction<?, ?> blockLoaderValueFunction() {
+            throw new UnsupportedOperationException("blockLoaderValueFunction is not supported");
```

Comment:
I think it'd be harder to make mistake with this if we didn't `default` implement it. Though I'd be happy with a "for testing" version of this thing that throws from every method so we don't have to change every single test in follow-up PRs.

Also! Can this return `null` if there's no function to apply?

Reply:
> I think it'd be harder to make mistake with this if we didn't default implement it.

Yep, my thinking was to reduce the blast radius for this at least at PoC level. I can remove the default when switching to ready for review 👍 

> Also! Can this return null if there's no function to apply?

I will do that, as I was going to signal this via the `FUNCTION` extract preference - but I understand your concerns there.